src/codegen/statements.rs

```diff
@@ -1289,10 +1307,45 @@ fn try_catch(
             _ => vartab.temp_anonymous(&Type::String),
         };
 
+        let buf = Expression::Variable {
+            loc: Codegen,
+            ty: Type::DynamicBytes,
+            var_no: error_ret_data_var,
+        };
+
+        // Expect the returned data to contain at least the selector + 1 byte of data
```

Comment:
Isn't this substrate specific code?
Wouldn't it be nice to guard it behind an if statement or perhaps a crate?

Reply:
It is guarded behind an if. The code panics on any other target than Polkadot.

> Wouldn't it be nice to guard it behind an if statement or perhaps a trait?

Error selectors on Solidity are 4 bytes. Why does this need to go behind an if or trait?

The decoding of the error data should really go into a trait. But the PR isn't about fixing error data decoding, it's about bubbling up reverts.

Reply:
The range check I added is there because the ABI encoder returns empty error data if it doesn't have enough data. But if the error data is less than 4 bytes, there is no way it matches `Error(string)`. But I just realized there might be another problem, I'm gonna look into it again tomorrow)

Reply:
Yeah the code for generating try catch seems broken. Also having #1449 in `main` will really help writing test cases here